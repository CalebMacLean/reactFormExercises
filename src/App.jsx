// Imports
import React from 'react';
import ToDoList from './ToDoList';

/** App Component
 * 
 * Renders: The ToDoList component
 */
const App = () => {
    return (
        <div>
            <ToDoList />
        </div>
    );
};

export default App;