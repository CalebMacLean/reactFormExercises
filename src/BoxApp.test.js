// Imports
import React from 'react';
import {render} from '@testing-library/react';
import App from './BoxApp';

// Test Suite
describe('App', () => {
    // Smoke Test
    it('renders without crashing', () => {
        render(<App />);
    });
    // Snapshot Test
    it('matches snapshot', () => {
        const {asFragment} = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});