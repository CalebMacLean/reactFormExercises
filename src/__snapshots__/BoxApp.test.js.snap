// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="BoxList"
    >
      <div
        class="BoxList-boxes"
      >
        <div
          class="Box"
          style="width: 100px; height: 100px; background-color: red;"
        >
          <button>
            X
          </button>
        </div>
      </div>
      <div
        class="BoxList-form"
      >
        <form
          class="NewBoxForm"
        >
          <label
            for="width"
          >
            Width:
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
          <label
            for="height"
          >
            Height:
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
          <label
            for="backgroundColor"
          >
            Background Color:
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            type="text"
            value=""
          />
          <button>
            Add a new box!
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
